lib/ansible/modules/database/postgresql/postgresql_user_obj_stat_info.py

```diff
@@ -0,0 +1,328 @@
+    schema: custom
+
+- name: Collect information about user tables and indexes in the acme database
+  postgresql_user_objects_info:
+    db: acme
+    filter: tables, indexes
+'''
+
+RETURN = r'''
+indexes:
+  description: User index statistics
+  returned: always
+  type: dict
+  sample: {"public": {"test_id_idx": {"idx_scan": 0, "idx_tup_fetch": 0, "idx_tup_read": 0, "relname": "test", "size": 8192, ...}}}
+tables:
+  description: User table statistics.
+  returned: always
+  type: dict
+  sample: {"public": {"test": {"analyze_count": 3, "n_dead_tup": 0, "n_live_tup": 0, "seq_scan": 2, "size": 0, "total_size": 8192, ...}}}
+functions:
+  description: User function statistics.
+  returned: always
+  type: dict
+  sample: {"public": {"inc": {"calls": 1, "funcid": 26722, "self_time": 0.23, "total_time": 0.23}}}
+'''
+
+try:
+    from psycopg2.extras import DictCursor
+except ImportError:
+    # psycopg2 is checked by connect_to_db()
+    # from ansible.module_utils.postgres
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.postgres import (
+    connect_to_db,
+    exec_sql,
+    get_conn_params,
+    postgres_common_argument_spec,
+)
+from ansible.module_utils.six import iteritems
+
+
+# ===========================================
+# PostgreSQL module specific support methods.
+#
+
+
+class PgUserObjStatInfo():
+    """Class to collect information about PostgreSQL user objects.
+
+    Args:
+        module (AnsibleModule): Object of AnsibleModule class.
+        cursor (cursor): Cursor object of psycopg2 library to work with PostgreSQL.
+
+    Attributes:
+        module (AnsibleModule): Object of AnsibleModule class.
+        cursor (cursor): Cursor object of psycopg2 library to work with PostgreSQL.
+        executed_queries (list): List of executed queries.
+        info (dict): Statistics dictionary.
+        obj_func_mapping (dict): Mapping of object types to corresponding functions.
+        schema (str): Name of a schema to restrict stat collecting.
+    """
+
+    def __init__(self, module, cursor):
+        self.module = module
+        self.cursor = cursor
+        self.info = {
+            'functions': {},
+            'indexes': {},
+            'tables': {},
+        }
+        self.obj_func_mapping = {
+            'functions': self.get_func_stat,
+            'indexes': self.get_idx_stat,
+            'tables': self.get_tbl_stat,
+        }
+        self.schema = None
+
+    def collect(self, filter_=None, schema=None):
+        """Collect statistics information of user objects.
+
+        Kwargs:
+            filter_ (list): List of subsets which need to be collected.
+            schema (str): Restrict stat collecting by certain schema.
+
+        Returns:
+            ``self.info``.
+        """
+        if schema:
+            self.set_schema(schema)
+
+        if filter_:
+            for obj_type in filter_:
+                self.obj_func_mapping.get(obj_type.strip())()
```

Comment:
I think this might explode if `obj_type` is not one of `functions`, `indexes` or `tables`, maybe it would be safer to do something like?
```suggestion
                obj_func = self.obj_func_mapping.get(obj_type.strip())
                if obj_func is not None:
                    obj_func()
                else:
                    module.warn("Unknown filter option '{}'".filter(obj_type.strip())
```

Reply:
done, thanks!